vmdk_plugin/drivers/vmdk/vmdkops/esx_vmdkcmd.go

```diff
@@ -90,15 +89,23 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
 	defer C.free(unsafe.Pointer(beS))
 
 	// Get the response data in json
-	ans := (*C.be_answer)(C.malloc(C.sizeof_struct_be_answer))
+	ans := (*C.be_answer)(C.calloc(C.sizeof_struct_be_answer, 0))
 	defer C.free(unsafe.Pointer(ans))
 
 	for i := 0; i <= maxRetryCount; i++ {
-		_, err = C.Vmci_GetReply(C.int(EsxPort), cmdS, beS, ans)
-		if err != nil {
-			var errno syscall.Errno
-			errno = err.(syscall.Errno)
-			msg := fmt.Sprintf("'%s' failed: %v (errno=%d).", cmd, err, int(errno))
+		var msg string
+		ret, err := C.Vmci_GetReply(C.int(EsxPort), cmdS, beS, ans)
+		if ret != 0 || err != nil {
+			if err != nil {
+				var errno syscall.Errno
+				errno = err.(syscall.Errno)
+				msg = fmt.Sprintf("'%s' failed: %v (errno=%d).", cmd, err, int(errno))
+				if errno == syscall.ECONNRESET || errno == syscall.ETIMEDOUT {
+					msg += " Cannot communicate with ESX, please refer to the FAQ https://github.com/vmware/docker-volume-vsphere/wiki#faq"
+				}
+			} else if ret != 0 {
```

Comment:
why not simply `else` ? I am not sure why `if ret != 0` is needed here. 
Or you can drop L98 and instead do `if err != nil {} ; if  ret != 0 {} (no `else` or `else if`)